### PR TITLE
utils_net: correct ipv6 address

### PR DESCRIPTION
**Pull request by XueqiangWei**
1. need to remove ifname.
  e.g. 'addr': 'fe80::af43:41fd:5437:9523%switch'
2. fix invalid syntax

ID: 1807888
Signed-off-by: Xueqiang Wei <xuwei@redhat.com>

### CODE REVIEW

**Comment by XueqiangWei** [comment]
(1/2) Host_RHEL.m8.u2.product_av.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.1.0.x86_64.io-github-autotest-qemu.uefishell.check_network.with_ipv4.q35: PASS (87.08 s)
 (2/2) Host_RHEL.m8.u2.product_av.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.1.0.x86_64.io-github-autotest-qemu.uefishell.check_network.with_ipv6.q35: PASS (142.01 s)

**Comment by XueqiangWei** [comment]
@zhencliu please help review it. Many thanks.

File: virttest/utils_net.py [hunk 1]
```diff
@@ -3390,17 +3390,17 @@ def get_ip_address_by_interface(ifname, ip_ver="ipv4", linklocal=False):
         try:
             if linklocal:
                 return [a['addr'] for a in addr
-                        if a['addr'].lower().startswith(linklocal_prefix)][0]
+                        if a['addr'].lower().startswith(linklocal_prefix)][0].split("%")[0]
```
Comment (PaulYuuu) [inline virttest/utils_net.py]: The current IPv6 address is in the correct format.
```python
>>> netifaces.ifaddresses("switch").get(netifaces.AF_INET6)
[{'addr': '2620:52:0:10c8:5e:c2cc:ece7:ae26', 'netmask': 'ffff:ffff:ffff:ffff::/64'}, {'addr': 'fe80::6946:86f8:e70d:8c7c%switch', 'netmask': 'ffff:ffff:ffff:ffff::/64'}]
```
IPv6 addresses with the prefix `fe80::/64` are link-local addresses that are constructed by combining that prefix with the hardware address of the network device, so the device ID should be reserved. And you can see the return format of `ping6` also include it.
```bash
# ping6 fe80::6946:86f8:e70d:8c7c%switch -c 4
PING fe80::6946:86f8:e70d:8c7c%switch(fe80::6946:86f8:e70d:8c7c%switch) 56 data bytes
64 bytes from fe80::6946:86f8:e70d:8c7c%switch: icmp_seq=1 ttl=64 time=0.017 ms
64 bytes from fe80::6946:86f8:e70d:8c7c%switch: icmp_seq=2 ttl=64 time=0.006 ms
64 bytes from fe80::6946:86f8:e70d:8c7c%switch: icmp_seq=3 ttl=64 time=0.006 ms
64 bytes from fe80::6946:86f8:e70d:8c7c%switch: icmp_seq=4 ttl=64 time=0.006 ms

# ping6 fe80::6946:86f8:e70d:8c7c -c 4
PING fe80::6946:86f8:e70d:8c7c(fe80::6946:86f8:e70d:8c7c) 56 data bytes
64 bytes from fe80::6946:86f8:e70d:8c7c%switch: icmp_seq=1 ttl=64 time=0.017 ms
64 bytes from fe80::6946:86f8:e70d:8c7c%switch: icmp_seq=2 ttl=64 time=0.006 ms
64 bytes from fe80::6946:86f8:e70d:8c7c%switch: icmp_seq=3 ttl=64 time=0.006 ms
64 bytes from fe80::6946:86f8:e70d:8c7c%switch: icmp_seq=4 ttl=64 time=0.006 ms
```

File: virttest/utils_net.py [hunk 2]
```diff
@@ -3390,17 +3390,17 @@ def get_ip_address_by_interface(ifname, ip_ver="ipv4", linklocal=False):
...
             else:
                 return [a['addr'] for a in addr
-                        if not a['addr'].lower().startswith(linklocal_prefix)][0]
+                        if not a['addr'].lower().startswith(linklocal_prefix)][0].split("%")[0]
         except IndexError:
             logging.warning("No IP address configured for "
-                            "the network interface %s !", ifname)
+                            "the network interface %s !" % ifname)
```
Comment (PaulYuuu) [inline virttest/utils_net.py]: It's not a typo.
Please refer to the documentation of the `logging` library.

File: virttest/utils_net.py (hunk 1 above)
Comment (XueqiangWei) [reply virttest/utils_net.py]: 1. tested it under uefi shell, it doesn't work.

FS0:\> ping6 -n 5 -s fe80::9834:39ff:fe31:a413 fe80::af43:41fd:5437:9523%switch 

Ping6: Invalid IP6 address, fe80::af43:41fd:5437:9523%switch
 
2. remove \"'%switch \", it works well. I will double check with edk2 developer whether fix it under uefi shell. thanks

FS0:\> ping6 -n 5 -s fe80::9834:39ff:fe31:a413 fe80::af43:41fd:5437:9523 
Ping fe80::af43:41fd:5437:9523 16 data bytes
16 bytes from fe80::af43:41fd:5437:9523 : icmp_seq=1 ttl=64 time0~9ms
16 bytes from fe80::af43:41fd:5437:9523 : icmp_seq=2 ttl=64 time0~9ms
16 bytes from fe80::af43:41fd:5437:9523 : icmp_seq=3 ttl=64 time0~9ms
16 bytes from fe80::af43:41fd:5437:9523 : icmp_seq=4 ttl=64 time0~9ms
16 bytes from fe80::af43:41fd:5437:9523 : icmp_seq=5 ttl=64 time0~9ms

5 packets transmitted, 5 received, 0% packet loss, time 0ms

Rtt(round trip time) min=0~9ms max=0~9ms avg=0~9ms

File: virttest/utils_net.py (hunk 2 above)
Comment (XueqiangWei) [reply virttest/utils_net.py]: I see,it's not a syntax error, but the output should have the variable value. If not update it, the output contains \"%s\". It's not friendly, so need to update it. thanks.

e.g.
before updated:
No IP address configured for the network interface %s !  switch

after updated:
No IP address configured for the network interface  switch !

Comment (PaulYuuu) [reply virttest/utils_net.py]:
```python
Python 3.7.1 (v3.7.1:260ec2c36a, Oct 20 2018, 14:57:15) [MSC v.1915 64 bit (AMD64)] on win32
Type "help", "copyright", "credits" or "license" for more information.
>>> import logging
>>> ifname = "switch"
>>> logging.error("Network interface: %s" % ifname)
ERROR:root:Network interface: switch
>>> logging.error("Network interface: %s", ifname)
ERROR:root:Network interface: switch
```

Comment (XueqiangWei) [reply virttest/utils_net.py]: OK, I will double check my debug logs. thanks.

**Comment by XueqiangWei** [comment]
Close this pullreq first. When getting feedback from edk2 developer, if it needs change, I will reopen it. Thanks.

**Pull request closed without merging**